gen/templates/generic/data_source.go

```diff
@@ -152,18 +158,37 @@ func (d *{{camelCase .Name}}DataSource) Read(ctx context.Context, req datasource
 		return
 	}
 
-	tflog.Debug(ctx, fmt.Sprintf("%s: Beginning Read", config.Id.String()))
 
-	res, err := d.client.Get("{{if .GetRestEndpoint}}{{.GetRestEndpoint}}{{else}}{{.RestEndpoint}}{{end}}" + config.Id.ValueString())
+	{{ if not .RemoveId}}
+	tflog.Debug(ctx, fmt.Sprintf("%s: Beginning Read", config.Id.String()))
```

Comment:
The debug statement should be outside of the "if" template block, as we want the log message to be created in any case. If the object does have a UUID we should be using something else that uniquely identifies it.

Reply:
Yea I think this gets back to the previous point on how we should set/structure/include and ID field